x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -159,87 +186,65 @@ impl Key {
     }
 }
 
-macro_rules! ffi_linker {
-    ($mod:literal, $link:literal, $caller:ident, $key:ident) => {
-        #[link(wasm_import_module = $mod)]
-        extern "C" {
-            #[link_name = $link]
-            fn ffi(caller: CPointer, key: CPointer) -> i32;
-        }
-
-        let $caller = to_ffi_ptr($caller.id())?;
-        let $key = to_ffi_ptr($key)?;
-    };
-    ($mod:literal, $link:literal, $caller:ident, $key:ident, $value:ident) => {
-        #[link(wasm_import_module = $mod)]
-        extern "C" {
-            #[link_name = $link]
-            fn ffi(caller: CPointer, key: CPointer, value: CPointer) -> i32;
-        }
-
-        let $caller = to_ffi_ptr($caller.id())?;
-        let $key = to_ffi_ptr($key)?;
-        let $value = to_ffi_ptr($value)?;
-    };
+#[derive(BorshSerialize)]
+pub(super) struct PutArgs {
+    pub(super) caller: Program,
+    pub(super) key: Vec<u8>,
+    pub(super) bytes: Vec<u8>,
 }
 
-macro_rules! call_host_fn {
-    (
-        wasm_import_module = $mod:literal
-        link_name = $link:literal
-        args = ($caller:ident, $key:ident)
-    ) => {{
-        ffi_linker!($mod, $link, $caller, $key);
-
-        unsafe { ffi($caller, $key) }
-    }};
-
-    (
-        wasm_import_module = $mod:literal
-        link_name = $link:literal
-        args = ($caller:ident, $key:ident, $value:ident)
-    ) => {{
-        ffi_linker!($mod, $link, $caller, $key, $value);
-
-        unsafe { ffi($caller, $key, $value) }
-    }};
+#[derive(BorshSerialize)]
+pub(super) struct GetAndDeleteArgs {
+    pub(super) caller: Program,
+    pub(super) key: Vec<u8>,
```

Comment:
Wrote this in a different module, didn't edit when I moved. 
```suggestion
struct GetAndDeleteArgs {
    caller: Program,
    key: Vec<u8>,
```